mutable_tree.go

```diff
@@ -378,14 +379,29 @@ func (tree *MutableTree) LoadVersion(targetVersion int64) (int64, error) {
 	return latestVersion, nil
 }
 
-// LoadVersionOverwrite returns the version number of targetVersion.
-// Higher versions' data will be deleted.
+// LoadVersionForOverwriting attempts to load a tree at a previously committed
+// version. Any versions greater than targetVersion will be deleted along with
+// their respective metadata.
 func (tree *MutableTree) LoadVersionForOverwriting(targetVersion int64) (int64, error) {
 	latestVersion, err := tree.LoadVersion(targetVersion)
 	if err != nil {
 		return latestVersion, err
 	}
-	tree.deleteVersionsFrom(targetVersion + 1) // nolint:errcheck
+
+	if err := tree.deleteVersionsFrom(targetVersion + 1); err != nil { // nolint
+		return latestVersion, err
+	}
+
+	vm, err := tree.ndb.GetVersionMetadata(targetVersion)
+	if err != nil {
+		return latestVersion, err
+	}
+
+	vm.Updated = time.Now().UTC().Unix()
```

Comment:
I don't think we need to update this, since this version isn't changing in any way?

Reply:
Indeed. I'll remove it.